openhands/runtime/base.py

```diff
@@ -360,12 +360,7 @@ async def clone_or_init_repo(
             return ''
 
         provider = repository.git_provider
-        provider_domains = {
-            ProviderType.GITHUB: 'github.com',
-            ProviderType.GITLAB: 'gitlab.com',
-        }
-
-        domain = provider_domains[provider]
+        domain = git_provider_tokens[provider].host
```

Comment:
```suggestion
        provider_domains = {
            ProviderType.GITHUB: 'github.com',
            ProviderType.GITLAB: 'gitlab.com',
        }
        domain = git_provider_tokens[provider].host if git_provider_tokens[provider].host else provider_domains[provider]
```

Host doesn't always exist so we should default to the provider domains above